app/controllers/bookings_controller.rb

```diff
@@ -1,8 +1,49 @@
 class BookingsController < ApplicationController
+  before_action :find_booking, only: [:show :edit, :update, :destroy]
+
+  def index
+    @bookings = Booking.all
+  end
 
   def new
+    @booking = Booking.new
   end
 
   def create
+    @booking = Booking.new(booking_params)
```

Comment:
@chihoko12 , how do you get the user and the boat in order to pass them to the Booking? I think according to the routes, the user is params[user_id]. We can't really check in the browser for now, I guess.

Reply:
i add authenticate_user for new and create methods and set @booing.user = take current_user.id.